File: 网站制作/frontend/src/types/device.ts
export type DeviceStatus = 'online' | 'offline' | 'busy' | 'error';
export type AlertType = 'paper_out' | 'paper_jam' | 'paper_low' | 'toner_low' | 'toner_empty' | 'offline' | 'service_required' | 'fuser_warning' | 'drum_low';
export type AlertSeverity = 'info' | 'warning' | 'critical';

export interface Printer {
  id: string;
  name: string;
  brand: string;
  model: string;
  serial_number: string | null;
  ip_address: string;
  mac_address: string | null;
  location: string | null;
  status: DeviceStatus;
  toner_level: number;
  toner_type: string | null;
  paper_level: number;
  total_pages_printed: number;
  firmware_version: string | null;
  supports_color: boolean;
  supports_duplex: boolean;
  max_paper_size: string;
  last_seen_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface PrinterStatusLog {
  id: string;
  printer_id: string;
  status: DeviceStatus;
  toner_level: number | null;
  paper_level: number | null;
  error_code: string | null;
  error_message: string | null;
  ip_address: string | null;
  response_time_ms: number | null;
  recorded_at: string;
}

export interface PrinterAlert {
  id: string;
  printer_id: string;
  alert_type: AlertType;
  severity: AlertSeverity;
  message: string;
  is_resolved: boolean;
  resolved_by: string | null;
  resolved_at: string | null;
  created_at: string;
}

export interface PrinterStats {
  total_pages: number;
  pages_today: number;
  pages_this_week: number;
  pages_this_month: number;
  total_errors: number;
  uptime_percentage: number;
}

export interface DeviceStatusUpdate {
  id: string;
  status: DeviceStatus;
  toner_level: number;
  paper_level: number;
  timestamp: string;
}

File: backend/app/schemas/device.py
from __future__ import annotations

from typing import Optional
from datetime import datetime
from uuid import UUID

from pydantic import Field

from app.schemas.common import ORMModel


class PrinterOut(ORMModel):
    """打印机完整信息响应"""

    id: UUID
    name: str
    brand: str
    model: str
    serial_number: Optional[str] = None
    ip_address: str
    mac_address: Optional[str] = None
    location: Optional[str] = None
    status: str
    toner_level: int = 0
    toner_type: Optional[str] = None
    paper_level: int = 0
    total_pages_printed: int = 0
    firmware_version: Optional[str] = None
    snmp_community: str = "public"
    snmp_port: int = 161
    supports_color: bool = False
    supports_duplex: bool = False
    max_paper_size: str = "A4"
    last_seen_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime


class PrinterStatusOut(ORMModel):
    """打印机实时状态响应"""

    id: UUID
    printer_id: UUID
    status: str
    toner_level: Optional[int] = None
    paper_level: Optional[int] = None
    error_code: Optional[str] = None
    error_message: Optional[str] = None
    ip_address: Optional[str] = None
    response_time_ms: Optional[int] = None
    recorded_at: datetime


class PrinterAlertOut(ORMModel):
    """打印机告警响应"""

    id: UUID
    printer_id: UUID
    alert_type: str
    severity: str
    message: str
    is_resolved: bool = False
    resolved_by: Optional[UUID] = None
    resolved_at: Optional[datetime] = None
    created_at: datetime


class PrinterStatsOut(ORMModel):
    """打印机统计信息响应"""

    total_pages: int = 0
    pages_today: int = 0
    pages_this_week: int = 0
    pages_this_month: int = 0
    total_errors: int = 0
    uptime_percentage: float = 100.0


class PrinterListQuery(ORMModel):
    """打印机列表查询参数"""

    brand: Optional[str] = None
    status: Optional[str] = None
    search: Optional[str] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)
